addons/account/models/partner.py

```diff
@@ -607,6 +631,35 @@ def _get_duplicated_bank_accounts(self):
         domain = expression.AND([domain, [('partner_id', '!=', self._origin.id)]])
         return self.env['res.partner.bank'].search(domain)
 
+    @api.depends('company_id')
+    def _compute_display_invoice_edi_format(self):
+        self.display_invoice_edi_format = len(self._fields['invoice_edi_format'].selection)
+
+    @api.depends('company_id')
+    def _compute_display_invoice_template_pdf_report_id(self):
+        available_templates_count = self.env['ir.actions.report'].search_count([('is_invoice_report', '=', True)], limit=2)
+        self.display_invoice_template_pdf_report_id = available_templates_count > 1
+
+    @api.depends_context('company')
+    def _compute_invoice_edi_format(self):
+        values = self.env['ir.property'].sudo()._get_multi('invoice_edi_format', self._name, self.ids)
+        for partner in self:
+            if partner.id not in values:
+                partner.invoice_edi_format = partner._get_suggested_invoice_edi_format()
+            else:
+                partner.invoice_edi_format = values[partner.id]
+
+    def _search_invoice_edi_format(self, operator, value):
+        return self._fields['invoice_edi_format']._search_company_dependent(self.with_company(self.env.company.root_id), operator, value)
```

Comment:
confused, the search and compuate are totally not equivalent. Since I have to refactor the code for jsonb company dependent, could you explain it a little?